File: Problems/C/53.c
int compare(int num1, int num2)
{
    if(num1 > num2)
        return num1;
    else
        return num2;
}

int maxSubArray(int* nums, int numsSize){
    int tempMax = nums[0];
    int max = nums[0];
        
    for(int i = 1 ; i < numsSize ; i++){
        tempMax = compare(nums[i], tempMax + nums[i]);
        max = compare(tempMax, max);
    }
        
    return max;
}

File: Problems/Python 3/53.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        tempMax = float('-inf')
        maxNum = 0
        
        for num in nums:
            if maxNum > 0:
                maxNum += num
            else:
                maxNum = num
            
            tempMax = max(tempMax, maxNum)
        
        return tempMax
